data/crawler/src/processing/downloader.py

```
@@ -0,0 +1,324 @@
+                else:
+                    logger.error(
+                        f"Failed to download {url} after {self.max_retries} attempts"
+                    )
+                    raise
+
+    def is_valid_url(self, url: str) -> bool:
+        """Check if URL is valid"""
+        try:
+            result = urlparse(url)
+            return all([result.scheme in ["http", "https"], result.netloc])
+        except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)